staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto

```diff
@@ -174,6 +174,38 @@ message PortMapping {
     string host_ip = 4;
 }
 
+// IpRanges specifies the ip ranges for address allocation of a sandbox
+message IpRanges {
```

Comment:
/assign @Random-Liu 

is v1alpha2 locked at this point?
i don't think we are incrementing the API version on new fields (which seems the correct way to manage the API).

Reply:
Is this an input to the CRI  for assigning Pod IP's ? I am confused , Without this, how did the Pod IP assignment happen currently when using CRI ?

Reply:
>The network address is just passing the PodCIDR dynamically to the plugin, so it no longer has to retrieve it from the APIServer.

I am confused about the above . Does the kubelet get the pod ip from apiserver ?

Reply:
this is an option to send directly the ipRanges field to the CNI (the address allocation range for pods), you can check a better description in https://github.com/containernetworking/cni/blob/master/CONVENTIONS.md.
Currently, you have to configure the address allocation range in your `/etc/cni/net.d/cni.conf`, and it has to match the `cluster-cidr` otherwise the pod communication doesn't work.
What some CNI plugins were doing is to retrieve the PodCIDR from the node to write it in the local config file, with this capability this is no longer needed because the CRI pass the address allocation range directly to the CNI plugin